beacon-chain/blockchain/service.go

```diff
@@ -18,31 +17,38 @@ var log = logrus.WithField("prefix", "blockchain")
 // ChainService represents a service that handles the internal
 // logic of managing the full PoS beacon chain.
 type ChainService struct {
-	validator                        bool
-	ctx                              context.Context
-	cancel                           context.CancelFunc
-	beaconDB                         *database.DB
-	chain                            *BeaconChain
-	web3Service                      *powchain.Web3Service
-	canonicalBlockEvent              chan *types.Block
-	canonicalCrystallizedStateEvent  chan *types.CrystallizedState
-	latestBeaconBlock                chan *types.Block
-	processedBlockHashes             [][32]byte
-	processedActiveStateHashes       [][32]byte
-	processedCrystallizedStateHashes [][32]byte
+	ctx                            context.Context
+	cancel                         context.CancelFunc
+	beaconDB                       *database.DB
+	chain                          *BeaconChain
+	web3Service                    *powchain.Web3Service
+	validator                      bool
+	canonicalBlockFeed             *event.Feed
+	canonicalCrystallizedStateFeed *event.Feed
+	latestProcessedBlock           chan *types.Block
+	lastSlot                       uint64
+	processedBlockHashes           [][32]byte
+	// These are the data structures used by the fork choice rule.
+	// We store processed blocks and states into a slice by SlotNumber.
+	// For example, at slot 5, we might have received 10 different blocks,
+	// and a canonical chain must be derived from this DAG.
+	//
+	// NOTE: These are temporary and will be replaced by a structure
```

Comment:
Can you create a github issue and reference that issue id in this note so we can track progress?

Reply:
Done in #429

Reply:
I meant to add that issue reference in this code so we can track it. I.e. if someone removes this comment because they have resolved the task, then we can see that this issue should be closed in the PR.